ui/src/clockface/components/resource_list/ResourceName.tsx

```diff
@@ -0,0 +1,160 @@
+// Libraries
+import React, {Component, KeyboardEvent, ChangeEvent, MouseEvent} from 'react'
+import classnames from 'classnames'
+
+// Components
+import {Input, ComponentSize} from 'src/clockface'
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+
+// Decorators
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+// Styles
+import 'src/clockface/components/resource_list/ResourceName.scss'
+
+interface PassedProps {
+  onUpdate: (name: string) => void
+  name: string
+  onEditName?: (e?: MouseEvent<HTMLAnchorElement>) => void
+  placeholder?: string
+  noNameString: string
+}
+
+interface DefaultProps {
+  parentTestID?: string
+  buttonTestID?: string
+  inputTestID?: string
+  hrefValue?: string
+}
+
+type Props = PassedProps & DefaultProps
+
+interface State {
+  isEditing: boolean
+  workingName: string
+}
+
+@ErrorHandling
+class ResourceName extends Component<Props, State> {
```

Comment:
Maybe we should just create generic inline edit components for this that aren't tied to the resource list. That way we can use them for name, description, and whatever else we need and they can be used outside of this list.

Reply:
I made an issue for this: https://github.com/influxdata/influxdb/issues/12171